kmip/services/kmip_client.py

```diff
@@ -859,3 +860,6 @@ def _set_variables(self, host, port, keyfile, certfile,
 
         self.password = conf.get_valid_value(
             password, self.config, 'password', conf.DEFAULT_PASSWORD)
+
+        self.timeout = conf.get_valid_value(
```

Comment:
The timeout being equal to zero is fine, that's an accepted state by the socket; it effectively puts the socket into non-blocking mode. I'm not sure what would happen on negative values; I can add a check for that.